### PR TITLE
binlog: allow multiple ddl targets

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Allow multiple DDL targets in binlog info. It's necessary for cdc to sync such DDLs (e.g. rename tables).

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: parser/model/ddl.go [hunk 1]
```diff
@@ -287,6 +301,15 @@ func (job *Job) FinishDBJob(jobState JobState, schemaState SchemaState, ver int6
 	job.BinlogInfo.AddDBInfo(ver, dbInfo)
 }
 
+// FinishDBJob is called when a job is finished.
+// It updates the job's state information and adds dbInfos the binlog.
+func (job *Job) FinishMultipleDBJob(jobState JobState, schemaState SchemaState, ver int64, dbInfos []*DBInfo) {
```
Comment (zimulala) [inline parser/model/ddl.go:306]: This function is not used at present, shall we remove it now?

File: parser/model/ddl.go [hunk 2]
```diff
@@ -175,6 +175,11 @@ type HistoryInfo struct {
 	DBInfo        *DBInfo
 	TableInfo     *TableInfo
 	FinishedTS    uint64
+
+	// MultipleDBInfo is like DBInfo but only for operations updating multiple DBs.
+	MultipleDBInfo []*DBInfo
```
Comment (zimulala) [inline parser/model/ddl.go:180]: This field is not used, could you remove it?

File: ddl/table.go [hunk 3]
```diff
@@ -798,21 +798,22 @@ func onRenameTables(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error
...
 		}
+		tblInfos = append(tblInfos, tblInfo)
 	}
 
 	ver, err = updateSchemaVersion(t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
-	job.FinishTableJob(model.JobStateDone, model.StatePublic, ver, tblInfo)
+	job.FinishMultipleTableJob(model.JobStateDone, model.StatePublic, ver, tblInfos)
```
Comment (overvenus) [inline ddl/table.go:816]: Will it cause breaking change if we remove `job.FinishTableJob`?

File: parser/model/ddl.go [hunk 4]
```diff
@@ -175,6 +175,9 @@ type HistoryInfo struct {
 	DBInfo        *DBInfo
 	TableInfo     *TableInfo
 	FinishedTS    uint64
+
+	// MultipleTableInfo is like TableInfo but only for operations updating multiple DBs.
+	MultipleTableInfo []*TableInfo
```
Comment (zimulala) [inline parser/model/ddl.go:180]: s/MultipleTableInfo/MultipleTableInfos?

File: parser/model/ddl.go [hunk 5]
```diff
@@ -175,6 +175,9 @@ type HistoryInfo struct {
 	DBInfo        *DBInfo
 	TableInfo     *TableInfo
 	FinishedTS    uint64
+
+	// MultipleTableInfo is like TableInfo but only for operations updating multiple DBs.
```
Comment (zimulala) [inline parser/model/ddl.go:179]:
```suggestion
	// MultipleTableInfo is like TableInfo but only for operations updating multiple tables.
```

File: ddl/table.go (hunk 3 above)
Comment (zimulala) [reply ddl/table.go:816]: Maybe we can do `h.TableInfo = tblInfo` in `FinishMultipleTableJob`. So, this shouldn't affect the old code much.

**Review by zimulala — COMMENTED** [review]
Maybe we need to update `Clean()` in `parser/model/ddl.go`.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by djshow832 — APPROVED** [review]

**Pull request merged**